plugins/basic/nexus-content-plugin/src/main/java/org/sonatype/nexus/content/internal/ContentAuthenticationFilter.java

```diff
@@ -71,7 +78,57 @@ protected AuthenticationToken createToken(final ServletRequest request, final Se
       return new ContentRestrictedToken(basis, request);
     }
     else {
+      AuthenticationToken token = createAuthenticationToken(request, response);
+      if (token != null) {
+        return token;
+      }
       return super.createToken(request, response);
     }
   }
+
+  @Override
+  protected boolean isAccessAllowed(final ServletRequest request, final ServletResponse response,
+                                    final Object mappedValue)
+  {
+    if (!isRestricted(request) && isLoginAttempt(request, response)) {
+      try {
+        return executeLogin(request, response) && super.isAccessAllowed(request, response, mappedValue);
+      }
+      catch (Exception e) {
+        throw Throwables.propagate(e);
+      }
+    }
+    return super.isAccessAllowed(request, response, mappedValue);
+  }
+
+  @Override
+  protected boolean isLoginAttempt(ServletRequest request, ServletResponse response) {
+    if (isRestricted(request)) {
+      return super.isLoginAttempt(request, response);
+    }
+    AuthenticationToken token = createAuthenticationToken(request, response);
+    return token != null || super.isLoginAttempt(request, response);
+  }
+
+  private AuthenticationToken createAuthenticationToken(ServletRequest request, ServletResponse response) {
```

Comment:
Both this and `NexusAuthenticationFilter` share `NexusHttpAuthenticationFilter` as a superclass, so there's scope for sharing this logic in a utility method rather than duplicating it in both places. Just a suggestion, shouldn't block merging this in as it stands.